dart: 2.0.0 -> 2.7.1 (stable) + 2.0.0 -> 2.8.0-dev.10.0 (dev)

GRBurst opened the pull request:
<!-- Nixpkgs has a lot of new incoming Pull Requests, but not enough people to review this constant stream. Even if you aren't a committer, we would appreciate reviews of other PRs, especially simple ones like package updates. Just testing the relevant package/service and leaving a comment saying what you tested, how you tested it and whether it worked would be great. List of open PRs: <https://github.com/NixOS/nixpkgs/pulls>, for more about reviewing contributions: <https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#chap-reviewing-contributions>. Reviewing isn't mandatory, but it would help out a lot and reduce the average time-to-merge for all of us. Thanks a lot if you do! -->
###### Motivation for this change
I recently built an app using `flutter`. I ran into multiple issues, one being that the current version of `dart` in nixos is not compatible with the newest `flutter` version. Currently, adding support for `flutter` is discussed in this issue #36759 and it looks promising to get added into the nixos ecosystem :-)

In order to be able to build the app, I had to update the dart version. There is another issue complaining about the old version of dart in #78183 and another pull request which is outdated again #60607

Please do not hesitate to reply, since I would be happy having this updated in the nixos packages.
 
###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

GRBurst commented:
@xeji since you merged the last pr of dart `dart: 1.24.3 -> 2.0.0`, I added you as reviewer ;-)

ericdallo reviewed (approved):
LGTM

GRBurst commented:
@Infinisil I do not know who is responsible for this to be merged, hence I checked the contributors of this file and saw that you contributed and are a member of Nixpkgs. Can you review and merge this please?

ericdallo commented on pkgs/development/interpreters/dart/default.nix:
```diff
@@ -72,6 +58,7 @@ stdenv.mkDerivation {
   meta = {
     platforms = [ "i686-linux" "x86_64-linux" ];
     homepage = https://www.dartlang.org/;
+    maintainers = with maintainers; [ grburst ];
```
@GRBurst, I think you forgot to use `maintainer` from `stdenv.lib`
```suggestion
    maintainers = with stdenv.lib.maintainers; [ grburst ];
```

ericdallo commented:
Please squash commits.

ericdallo commented:
@GRBurst, I mean squash all PR commits into single one `dart: 2.7.0 -> 2.7.1 and 2.0.0-dev.26.0 -> 2.8.0-dev.5.0`. [This way](https://www.internalpointers.com/post/squash-commits-into-one-git) makes things cleaner and easy to read on master commit history :sweat_smile:

GRBurst commented:
@ericdallo yes you are right. I also updated dart dev version to dev.9.0 (and successfully repeated all test with that version).

GRBurst commented:
@Infinisil @xeji please, please review or suggest an appropriate reviewer

GRBurst commented:
@worldofpeace since I have positive experiences with you as a reviewer and your response time I kindly ask you to review or suggest an appropriate reviewer for this package update

worldofpeace commented:
@GrahamcOfBorg build dart dart_dev dart_stable dart_old

worldofpeace reviewed (changes requested):
You need to update the versions in all-packages.nix, or I can't build the dev versions etc.
```

  dart = callPackage ../development/interpreters/dart { };
  dart_stable = dart.override { version = "2.0.0"; };
  dart_old = dart.override { version = "1.24.3"; };
  dart_dev = dart.override { version = "2.0.0-dev.26.0"; };
```

it'll fail with `error: unsupported version/system: 2.0.0-dev.26.0/x86_64-linux` for example on `dart_dev`.

worldofpeace commented:
> @worldofpeace since I have positive experiences with you as a reviewer and your response time I kindly ask you to review or suggest an appropriate reviewer for this package update

I'm nixos's first responder :sparkling_heart:

GRBurst commented:
> You need to update the versions in all-packages.nix, or I can't build the dev versions etc.
> 
> ```
> 
>   dart = callPackage ../development/interpreters/dart { };
>   dart_stable = dart.override { version = "2.0.0"; };
>   dart_old = dart.override { version = "1.24.3"; };
>   dart_dev = dart.override { version = "2.0.0-dev.26.0"; };
> ```
> 
> it'll fail with `error: unsupported version/system: 2.0.0-dev.26.0/x86_64-linux` for example on `dart_dev`.

Good to have you on board :-)
I see - did not updated `all-packages.nix` after the updates appropriately 👍

GRBurst committed:
dart: 2.0.0 -> 2.7.1 (stable) + 2.0.0 -> 2.8.0-dev.10.0 (dev)

- dart stable: make it default
- add aarch64 (arm64) packages
- set myself as maintainer

GRBurst commented:
@GrahamcOfBorg build dart dart_dev dart_stable dart_old

GRBurst commented:
@GrahamcOfBorg build dart dart_dev dart_stable dart_old

worldofpeace commented on pkgs/top-level/all-packages.nix:
```diff
@@ -25783,9 +25783,9 @@ in
-  dart_dev = dart.override { version = "2.0.0-dev.26.0"; };
+  dart_old = dart.override    { version = "1.24.3";         };
+  dart_stable = dart.override { version = "2.7.1";          };
+  dart_dev = dart.override    { version = "2.8.0-dev.10.0"; };
```
I know this looks nice, but it's better if editors just vertically aligned in the buffer.
Could you revert it? It's not a codestyle we have.

GRBurst replied:
ah okay - sure thing

worldofpeace commented:
@GrahamcOfBorg build dart dart_dev dart_stable dart_old

worldofpeace commented:
Thanks @GRBurst :sparkles:
